larcore/Geometry/Geometry_service.cc

```diff
@@ -162,5 +165,98 @@ namespace geo {
 
   } // Geometry::LoadNewGeometry()
 
+  //......................................................................
+  void Geometry::FillGeomeryConfigurationInfo(fhicl::ParameterSet const& config)
+  {
+    
+    sumdata::GeometryConfigurationInfo confInfo;
+    confInfo.dataVersion = sumdata::GeometryConfigurationInfo::DataVersion_t{1};
+    
+    // version 1:
+    confInfo.detectorName = DetectorName();
+    
+    // all versions
+    confInfo.geometryServiceConfiguration = config.to_indented_string();
+    fConfInfo = std::move(confInfo);
+    
+    MF_LOG_TRACE("Geometry")
+      << "Geometry configuration information:\n" << fConfInfo;
+    
+  } // Geometry::FillGeomeryConfigurationInfo()
+
+  //......................................................................
+  bool Geometry::CheckConfigurationInfo
+    (sumdata::GeometryConfigurationInfo const& other) const
+  {
+    
+    MF_LOG_DEBUG("Geometry") << "New geometry information:\n" << other;
+    
+    return CompareConfigurationInfo(fConfInfo, other);
+    
+  } // Geometry::CheckConfigurationInfo()
+  
+  //......................................................................
+  sumdata::GeometryConfigurationInfo const& Geometry::ReadConfigurationInfo
+    (art::Run const& run)
+  {
+    
+    try {
+      return run.getByLabel<sumdata::GeometryConfigurationInfo>
+        (art::InputTag{"GeometryConfigurationWriter"});
```

Comment:
Should this be a `getValidHandle`?  I expect that `getByLabel` would never fail unless there is corrupt data.

Reply:
If the question is whether `getByLabel()` (overload with a single input tag as argument) throws when there is no product, from the unit tests it appears to do it, big time.
I did not use `getValidHandle()` because here I don't care about the handle but about the data product (hint is that if I used `getValidHandle()`, I would need further processing to `return` what I need to).
I might be missing your point though.

Reply:
Ah, I missed that the one-argument `getByLabel` overload was being used.  Sorry about that.